### PR TITLE
Added more tests for install-section parser to capture its behaviour

### LINKED ISSUE #66: module.yaml parsers: install-section parser is not covered by tests well
It is not obvious how install-section parsing works:
- how install-sections are inherited
- what is the purpose of `InstallData` class and its properties:
  * how exactly these properties are set
  * how these properties are affected by 'nuget' (#15) and 'external modules' features

This information is not covered either by [docs](https://github.com/skbkontur/cement/blob/f98b4f2e70b883a3933f55cf913503eb2cee3061/README-module.yaml.md) or by the unit tests.

There should be a better test coverage for install-section parsing as it is not trivial and can conceal logical problems (#67)

### LINKED ISSUE #67: install: nuget packages leak into InstallData properties
Consider a `module.yaml` example:

```
full-build:
  artifacts:
    - a1
    - nuget SomeNuget1
```

After calling `InstallYamlParser.Get()` I get a InstallData with these properties:
- Artifacts:
  * a1
  * nuget SomeNuget1  
- MainConfigBuildFiles:
  * a1
  * nuget SomeNuget1

It seems, that `nuget SomeNuget1` should not be either in `Artifacts` or in `MainConfigBuildFiles` collections. 

P.S. It's probably a bug, that appeared after merge request #15: [source1](https://github.com/skbkontur/cement/blob/cc24d78c06a42ffaf1f28357aa03df25e9c37aef/Common/YamlParsers/InstallYamlParser.cs#L41), [source2](https://github.com/skbkontur/cement/blob/cc24d78c06a42ffaf1f28357aa03df25e9c37aef/Common/YamlParsers/InstallYamlParser.cs#L48)

### PR DESCRIPTION
**Pull request by obiwan-smirnobi**
Added separate test cases for parsing artifacts/artefacts, build files, external modules, nuget packages, and MainConfigBuildFiles.

This pull request fixes #66
This pull request fixes #67

### CODE REVIEW

File: Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserMainConfigBuildFiles.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  install:
+    - file1
+    - file2
+    - file3
+    - module SomeModule3
+",
+                    new [] { "file1", "file2", "file3", "file4", "file5" })
+                .SetName("Install section: MainConfigBuildFiles. 'modules' are not leaked into MainConfigBuildFiles."),
+
+            // possibly a bug
```
Comment (KungA) [inline Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserMainConfigBuildFiles.cs]: fix it?

**Comment by KungA** [comment]
Please fix #67 here, and write short description xml-docs for `InstallData` properties.

File: Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserArtifacts.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    - file1
+    - module SomeModule
+    - nuget SomeNuget
+
+  artifacts:
+    - a1",
+                    new[] { "file1", "a1" })
+                .SetName("Install section: artifacts. Build files from install section are added to artifacts (ignoring 'module' and 'nuget' directive)."),
+
+            // possibly a bug
```
Comment (KungA) [inline Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserArtifacts.cs]: and here

File: Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserMainConfigBuildFiles.cs (hunk 1 above)
Comment (obiwan-smirnobi) [reply Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserMainConfigBuildFiles.cs]: done

File: Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserArtifacts.cs (hunk 2 above)
Comment (obiwan-smirnobi) [reply Tests/ParsersTests/YamlInstallSection/TestInstallYamlParserArtifacts.cs]: done

**Comment by obiwan-smirnobi** [comment]
> Please fix #67 here, and write short description xml-docs for InstallData properties.

I haven't fully grasped the meaning of each property of `InstallData` yet. Let's do it a bit later?

**Comment by KungA** [comment]
> > Please fix #67 here, and write short description xml-docs for InstallData properties.
> 
> I haven't fully grasped the meaning of each property of `InstallData` yet. Let's do it a bit later?

Let's do it now: tests should help us (what we test, if definition is unknown? :)).

- MainConfigBuildFiles: install of current configuration
- BuildFiles: install files of current and parent configuration (does it include artifacts? no test for this case)
- ExternalModules: install modules of current and parent configurations (seems not include install of referenced module)
- NuGetPackages: install nuget of current and parent configurations
- Artifacts: install and artifacts of current and parent configurations

**Comment by KungA** [comment]
I have also renamed properties to meaningful names 087a819eb0fccdaddd284a0056fbbc6afc18c8b1.

**Pull request merged**